src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.MultiSegment.cs

```diff
@@ -1108,6 +1109,7 @@ private bool TryGetNumberMultiSegment(ReadOnlySpan<byte> data, out int consumed)
             // TODO: https://github.com/dotnet/corefx/issues/33294
             Debug.Assert(data.Length > 0);
 
+            _numberHasExponent = false;
```

Comment:
I believe this is true, but I want to confirm: Is it guaranteed that the value of `_numberHasExponent` will only be relevant (and only used) when the JSON token is a number? That is, do we ever need to reset it to false for other tokens (strings, start object, etc.) or is setting it to false only when we see a number, sufficient?

Reply:
Read access to `_numberHasExponent` appears only in the methods capable of parsing a number with an exponent (`TryGetSingleValue`, `TryGetDoubleValue` and `TryGetDecimalValue`). Those methods only succeed if `TokenType == JsonTokenType.Number`. This in turn is only `true` after a successful call to either `TryGetNumber` or `TryGetNumberMultiSegment`, which is where `_numberHasExponent` is reset to `false`. 
The other public `TryGet***Value` methods either don't care abot exponents or call `Utf8Parser.TryParse` overloads that do not accept numbers with exponents anyways.

So, yes: `_numberHasExponent` should always have the right value when it is relevant. But it may not be correct (strictly speaking) at times when it does not matter (e.g. when parsing a string after parsing a number with an exponent).

(BTW: I find it somewhat counter-intuitive, that `Utf8JsonReader.TryGetInt[32|64]Value` returns false for `1E5` which could be considered an integer. But I assume that is by design as `int.TryParse("1E5")` also returns false, unless `NumberStyles.AllowExponent` is given).